internal/guest/runtime/runc/container.go

```diff
@@ -389,6 +400,70 @@ func (c *container) startProcess(
 		}
 	}
 
+	// This is to enabling container logging in Bare Metal Containers. This is in experimental stage now and need to be revisited.
+	var stdoutFifoPipe, stderrFifoPipe *os.File
+	if annotations != nil {
+		pipeNameSuffix, exists := annotations["io.microsoft.bmc.logging.pipelocation"]
+		pipeDirectory := "/run/gcs/containerlogs/"
+		stdoutPipeName := pipeDirectory + pipeNameSuffix + "-stdout"
+		stderrPipeName := pipeDirectory + pipeNameSuffix + "-stderr"
+		os.MkdirAll(pipeDirectory, os.ModePerm)
+		if exists {
+			_, err = os.Stat(stdoutPipeName)
+			if err != nil {
+				// fifo pipe does not exist, create one
+				err = syscall.Mkfifo(stdoutPipeName, 0666)
+				if err != nil {
+					msg := "Error creating named pipe:" + fmt.Errorf("outer error context: %w", err).Error()
+					logrus.Infof(msg)
+					return nil, errors.Wrapf(err, msg)
```

Comment:
For the error just do `fmt.Errorf("creating fifo %s: %w", stdoutPipeName, err)` and return that. Also not worth logging here since we are returning the error anyways.